boards/arm/rddrone_fmuk66/board.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2022, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/init.h>
+#include <zephyr/device.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+
+LOG_MODULE_REGISTER(rddrone_fmuk66_board_init);
+
+/* Initialization chain of rddrone fmuk66 board requires some delays before on board sensors
+ * could be accessed after power up. In particular bme680 and bmm150 sensors require,
+ * respectively 2ms and 1ms power on delay. In order to avoid delays sum, common delay is
+ * introduced in the board start up file. Below asserts ensure correct initialization order:
+ * on board regulators, board init (this), sensors init.
+ */
+#if defined(CONFIG_REGULATOR_FIXED)
+BUILD_ASSERT(CONFIG_RDDRONE_FMUK66_INIT_PRIORITY > CONFIG_REGULATOR_FIXED_INIT_PRIORITY,
+	"CONFIG_RDDRONE_FMUK66_INIT_PRIORITY must be higher than CONFIG_REGULATOR_FIXED_INIT_PRIORITY");
+#endif
+
+#if defined(CONFIG_SENSOR)
+BUILD_ASSERT(CONFIG_RDDRONE_FMUK66_INIT_PRIORITY < CONFIG_SENSOR_INIT_PRIORITY,
+	"CONFIG_RDDRONE_FMUK66_INIT_PRIORITY must be less than CONFIG_SENSOR_INIT_PRIORITY");
+#endif
+
+
+
+static int setup()
+{
+
+	if (IS_ENABLED(CONFIG_SENSOR)) {
+		/* Initialization chain of rddrone fmuk66 board requires some delays before on board
+		 * sensors could be accessed after power up. In particular bme680 and bmm150
+		 * sensors require, 2ms and 1ms power on delay respectively. In order not to sum
+		 * delays, common delay is introduced in the board start up file. This code is
+		 * executed after sensors are powered up and before their initialization.
+		 * It's ensured by build asserts at the beginning of this file.
+		 */
+		k_msleep(2);
+	}
```

Comment:
this seems like a workaround for bus voltage stabilization, why not use `startup-delay-us` in the regulator?

Reply:
Will revert on this very shortly

Reply:
@gmarull I have verified that startup-delay-us does not resolve the i2c sensor issue on rddrone board
I can further compare the effect of the delay I added vs startup-delay-us later, but for now please accept this change.

Reply:
Why doesn't `startup-delay-us` work? Isn't that what it's for?